src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -57,42 +57,54 @@ auto CMitsubaMaterialCompilerFrontend::getTexture(const CElementTexture* _elemen
 }
 
 
-auto CMitsubaMaterialCompilerFrontend::getEmissionProfile(const CElementEmissionProfile* _element) const -> tex_ass_type
+auto CMitsubaMaterialCompilerFrontend::getEmissionProfile(const CElementEmissionProfile* _element) const -> emission_profile_type
 {
     asset::IAsset::E_TYPE types[2]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+
     const auto key = SContext::emissionProfileCacheKey(*_element);
     auto viewBundle = m_loaderContext->override_->findCachedAsset(key, types, m_loaderContext->inner, 0u);
     if (!viewBundle.getContents().empty())
     {
         auto view = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(viewBundle.getContents().begin()[0]);
-
         types[0] = asset::IAsset::ET_SAMPLER;
-        const std::string samplerKey = m_loaderContext->samplerCacheKey(m_loaderContext->emissionProfileSamplerParams(*_element));
+        const auto& meta = *static_cast<const asset::CIESProfileMetadata*>(viewBundle.getMetadata());
+        const std::string samplerKey = m_loaderContext->samplerCacheKey(m_loaderContext->emissionProfileSamplerParams(*_element, meta));
         auto samplerBundle = m_loaderContext->override_->findCachedAsset(samplerKey, types, m_loaderContext->inner, 0u);
         assert(!samplerBundle.getContents().empty());
         auto sampler = core::smart_refctd_ptr_static_cast<asset::ICPUSampler>(samplerBundle.getContents().begin()[0]);
 
-        return { view, sampler, 1.0f };
+        return { view, sampler, &meta };
     }
-    return { nullptr, nullptr, 1.0f };
+    return { nullptr, nullptr, nullptr };
 }
 
 CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend::createEmitterNode(asset::material_compiler::IR* ir, const CElementEmitter* _emitter, core::matrix4SIMD transform) {
     assert(_emitter->type == CElementEmitter::AREA);
     auto* res = ir->allocNode<asset::material_compiler::IR::CEmitterNode>();
+    res->intensity = _emitter->area.radiance;
     if (_emitter->area.emissionProfile) {
         asset::material_compiler::IR::CEmitterNode::EmissionProfile profile;
-        asset::material_compiler::IR::INode::STextureSource tex;
-        std::tie(tex.image, tex.sampler, tex.scale) = getEmissionProfile(_emitter->area.emissionProfile);
-        assert(tex.image && "emission profile not loaded");
-        profile.texture = tex;
-        profile.normalizeEnergy = _emitter->area.emissionProfile->normalizeEnergy;
-        auto inverseTransform = core::transpose(core::concatenateBFollowedByA(transform, _emitter->area.emissionProfile->transform.matrix));
-        profile.left = inverseTransform[0];
-        profile.up = inverseTransform[1];
+        auto [image, sampler, meta] = getEmissionProfile(_emitter->area.emissionProfile);
+        if (!image) {
+            os::Printer::log("ERROR: Emission profile not loaded", ELL_ERROR);
+            return nullptr;
```

Comment:
that doesn't provide a default, it actually murders my emitter completely, please fallback to produce the same result as if `_emitter->area.emissionProfile` was null

Reply:
this path wouldnt be run if emisison profile is null right?

Reply:
oh nvm I misread your comment. I can just make this warning and keep going.

Reply:
yep, and uniform profile (no profile) but still having an emitter